relayer/src/supervisor.rs

```diff
@@ -379,3 +435,42 @@ pub fn collect_events(batch: EventBatch) -> CollectedEvents {
 
     collected
 }
+
+// TODO: Memoize this result
+fn get_counterparty_chain(
+    src_chain: &dyn ChainHandle,
+    src_channel_id: &ChannelId,
+    src_port_id: &PortId,
+) -> Result<ChainId, BoxError> {
+    info!(
+        chain_id = %src_chain.id(),
+        src_channel_id = %src_channel_id,
+        src_port_id = %src_port_id,
+        "getting counterparty chain"
+    );
+
+    use ibc::ics02_client::client_state::ClientState;
+
+    let src_channel = src_chain.query_channel(src_port_id, src_channel_id, Height::zero())?;
+
+    // TODO: Check channel state?
```

Comment:
Shall we fix the TODO or is this for later?

Reply:
It's actually more of a question than a TODO, was hoping you or @ancazamfir could advise me as to what to do here.

Reply:
If the channel doesn't exist, this query returns an `Uninitialized` channel with zero connection hops. So we will see an error when we try to get the connection further down.